src/Nancy/Xml/XmlConfigurationExtensions.cs

```diff
@@ -0,0 +1,24 @@
+﻿namespace Nancy.Xml
+{
+    using System.Text;
+    using Nancy.Configuration;
+
+    /// <summary>
+    /// Contains <see cref="XmlConfiguration"/> configuration extensions for <see cref="INancyEnvironment"/>.
+    /// </summary>
+    public static class XmlConfigurationExtensions
+    {
+        /// <summary>
+        /// Configures <see cref="XmlConfiguration"/>.
+        /// </summary>
+        /// <param name="environment"><see cref="INancyEnvironment"/> that should be configured.</param>
+        /// <param name="enableEncoding"><see langword="true"/> if encoding should be enabled, otherwise <see langword="false"/>.</param>
+        /// <param name="defaultEncoding">The default <see cref="Encoding"/>.</param>
+        public static void Xml(this INancyEnvironment environment, bool enableEncoding, Encoding defaultEncoding = null)

```

Comment:
This is per design and per the discussions we had during the creation of the config API. The idea is to promote the use of named parameters when calling these methods to make the configuration more explicit and clear

```c#
env.Foo.(10, "John Doe", new DateTime(2015, 11, 18), () => Guid.NewGuid());

vs.

env.Foo(
   maxNumberOfRequests: 10,
   defaultUserName: "John Doe",
   earliestDate: new Date(2015, 11, 18),
   userIdFactory: () => Guid.NewGuid()
);
```

It is already in place for the `DiagnosticsConfiguration` as well. Because this is a config API it adds a lot of clarity when reading the config code.